File: lab1/lab1_sharp/lab1_sharp/Worker.cs
using System.Threading;
using System;

namespace lab1_sharp
{
    public class Worker 
    {
        private readonly int _id;
        private readonly SuperThreadBreaker _threadBreaker;
        private readonly int _step;

        public Worker(int id, int step, SuperThreadBreaker breaker)
        {
            _id = id;
            _step = step;
            _threadBreaker = breaker;
        }

        public void Run()
        {        
            new Thread(() => 
            {
                long sum = 0;
                long count_steps = 0;
                bool isStopped = false;
                do
                {
                    count_steps++;
                    sum += _step;
                    isStopped = _threadBreaker.CanBreak(_id);
                } while (!isStopped);

                Console.WriteLine($"Id: {_id} | count_steps: {count_steps} | sum: {sum}");
            }).Start();        
        }
    }
}

File: lab1/task1/src/Worker.java
public class Worker extends Thread
{
    private final int _id;
    private final ThreadBreaker _threadBreaker;

    public Worker(int id, int lifeTimeInSeconds)
    {
        _id = id;
        _threadBreaker = new ThreadBreaker(lifeTimeInSeconds * 1000);
    }

    @Override
    public void run()
    {
        (new Thread(_threadBreaker)).start();
        long sum = 0;
        boolean isStopped = false;
        do
        {
            sum++;
            isStopped = _threadBreaker.CanBreak();
        } while (!isStopped);
        System.out.println("id: " + _id + " - " + sum);
    }
}
